src/video/allegro_v.cpp

```diff
@@ -139,52 +139,52 @@ static void GetVideoModes()
 	 * cards ourselves... and we need a card to get the modes. */
 	set_gfx_mode(_fullscreen ? GFX_AUTODETECT_FULLSCREEN : GFX_AUTODETECT_WINDOWED, 640, 480, 0, 0);
 
+	_resolutions.clear();
+
 	GFX_MODE_LIST *mode_list = get_gfx_mode_list(gfx_driver->id);
 	if (mode_list == nullptr) {
-		memcpy(_resolutions, default_resolutions, sizeof(default_resolutions));
-		_num_resolutions = lengthof(default_resolutions);
+		for (uint n = 0; n < lengthof(default_resolutions; n++) _resolutions.push_back(default_resolutions[n]);
 		return;
 	}
 
 	GFX_MODE *modes = mode_list->mode;
 
-	int n = 0;
 	for (int i = 0; modes[i].bpp != 0; i++) {
 		uint w = modes[i].width;
 		uint h = modes[i].height;
 		if (w >= 640 && h >= 480) {
-			int j;
-			for (j = 0; j < n; j++) {
+			uint j;
```

Comment:
whole block could be replaced with
Dimension might need an `operator==`

```suggestion
			bool has_resolution = std::find(_resolutions.begin(), _resolutions.end(), Dimension(w, h)) == _resolutions.end();
			if (!has_resolution) _resolutions.emplace_back(w, h);
```

Reply:
and a non default constructor too ;)